pkg/internal/controller/controller.go

```diff
@@ -130,6 +130,9 @@ func (c *Controller) Start(stop <-chan struct{}) error {
 	c.Queue = c.MakeQueue()
 	defer c.Queue.ShutDown() // needs to be outside the iife so that we shutdown after the stop channel is closed
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I think `TODO` is better suited since we want to eventually change the `Runnable` signature to take a context and then use that context here